src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -190,13 +201,25 @@ public function removeEventSubscriber(EventSubscriber $subscriber): void
     private function initializeListeners(string $eventName)
     {
         $this->initialized[$eventName] = true;
+
+        // We'll refill the whole array in order to keep the same order
+        $listeners = [];
```

Comment:
If I just do
```
$this->listeners[$eventName][$newHash] = $listener;
```
the order of the listeners change.

I'm not sure if there is a better solution to keep the order

Reply:
seems ok to be to just re-create it :man_shrugging: 

But maybe it would be good to add a test that makes sure we keep the order?

Reply:
The test already exist. If I don't keep the same order, multiple tests are failing because of the `assertSame` checks